In [1]:
# Import Libraries

import pandas as pd
import os
import numpy as np
import datetime
import matplotlib.pyplot as plt
import seaborn as sns
import mpu

%matplotlib inline

In [58]:
All_2019_data=pd.read_csv('D:\\GA Work_Essia\\Data\\Yearly data\\2019\\All_2019_data.csv')
All_2019_data = All_2019_data.drop_duplicates(['start_station'], keep='first')
Station_2019=All_2019_data[['start_station']]
Station_2019['start_station'].nunique()

234

In [59]:
All_2018_data=pd.read_csv('D:\\GA Work_Essia\\Data\\Yearly data\\2018\\All_2018_data.csv')
All_2018_data = All_2018_data.drop_duplicates(['start_station'], keep='first')
Station_2018=All_2018_data[['start_station']]
Station_2018['start_station'].nunique()

132

In [60]:
All_2017_data=pd.read_csv('D:\\GA Work_Essia\\Data\\Yearly data\\2017\\All_2017_data.csv')
All_2017_data = All_2017_data.drop_duplicates(['start_station'], keep='first')
Station_2017=All_2017_data[['start_station']]
Station_2017['start_station'].nunique()

133

In [61]:
All_2016_data=pd.read_csv('D:\\GA Work_Essia\\Data\\Yearly data\\2016\\All_2016_data.csv')
All_2016_data = All_2016_data.drop_duplicates(['start_station'], keep='first')
Station_2016=All_2016_data[['start_station']]
Station_2016['start_station'].nunique()

63

In [67]:
comparison_df = Station_2016.merge(Station_2017,
                              indicator=True,
                              how='outer')
comparison_df['_merge'].value_counts()

right_only    71
both          62
left_only      2
Name: _merge, dtype: int64

**Total Stations In 2016=63**

**Total Stations In 2017=133**

**Old 2016 Stations in 2017=62**

**New Stations in 2017=71**

**Shut down station in 2016=2**

In [69]:
comparison_df = Station_2017.merge(Station_2018,
                              indicator=True,
                              how='outer')
comparison_df['_merge'].value_counts()

both          132
left_only       1
right_only      0
Name: _merge, dtype: int64

**Total Stations In 2017=133**

**Total Stations In 2018=132**

**Old 2017 Stations in 2018=132**

**New Stations in 2018=0**

**Shut down station in 2016=1**

In [68]:
comparison_df = Station_2016.merge(Station_2018,
                              indicator=True,
                              how='outer')
comparison_df['_merge'].value_counts()

right_only    71
both          61
left_only      3
Name: _merge, dtype: int64

In [70]:
comparison_df = Station_2018.merge(Station_2019,
                              indicator=True,
                              how='outer')
comparison_df['_merge'].value_counts()

right_only    137
both           97
left_only      35
Name: _merge, dtype: int64

**Total Stations In 2018=132**

**Total Stations In 2019=234**

**Old 2018 Stations in 2019=97**

**New Stations in 2018=137**

**Shut down station in 2016=35**